src/views/product-downloads-view/types.ts

```diff
@@ -1,40 +1,77 @@
-import { ReleasesAPIResponse } from 'lib/fetch-release-data'
+import { ReleasesAPIResponse, GeneratedProps } from 'lib/fetch-release-data'
 import { VersionContextSwitcherProps } from 'components/version-context-switcher'
 import { SidecarMarketingCardProps } from './components/sidecar-marketing-card'
+import { CollectionCardPropsWithId } from 'components/collection-card'
+import { TutorialCardPropsWithId } from 'components/tutorial-card'
 
-export interface FeaturedTutorial {
-  description: string
-  href: string
-  title: string
-}
-
-export interface GroupedPackageManagers {
-  [os: string]: PackageManager[]
-}
+/**
+ * Raw page content, as structured in .json files
+ */
+export type FeaturedTutorialContent =
+  | {
+      collectionSlug: string
+      tutorialSlug: never
+    }
+  | {
+      tutorialSlug: string
+      collectionSlug: never
+    }
```

Comment:
I think these kinds of types are so fun! 🙌🏻

🚧 There is one thing I think needs updated, and maybe we're not seeing an error pop up because the content is coming from a JSON file and not a TS file. I'm fairly certain the properties set to `never` also need to be optional.

🎨 Would also prefer the properties to be alphabetical in both parts, but totally up to you!

```suggestion
/**
 * Raw page content, as structured in .json files
 */
export type FeaturedTutorialContent =
  | {
      collectionSlug: string
      tutorialSlug?: never
    }
  | {
      collectionSlug?: never
      tutorialSlug: string
    }
```